### PR TITLE
[sdk-metrics] Support .NET 9 Synchronous Gauge

### LINKED ISSUE #4805: Support .NET 9 Synchronous Gauge
# Feature Request
It would be great for synchronous gauge support to be added, now that it has been added to Experimental spec.

# Is your feature request related to a problem?
Use cases are described here: https://github.com/open-telemetry/opentelemetry-specification/issues/2318.

# Describe the solution you'd like
I would expect to have a method in the class Meter called CreateObservableGauge<T>() along with the class Gauge<T>. They should behave like the ObservableGauge instrument, but being synchronous instead of asynchronous.

This is described as an experimental feature in the OpenTelemetry specification: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#gauge

# Describe alternatives you've considered
Both UpDownCounter and ObservableGauge do not work for my usecases.

# Additional context
-

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
Fixes #4805
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

Updated the `Metric` class to add support for synchronous gauge.

Follow up: Update document - #5868

## Merge requirement checklist

* [X] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [X] Unit tests added/updated
* [X] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM, would you also update the docs? For example https://github.com/open-telemetry/opentelemetry-dotnet/blob/cc0b9e9d0f874d4cbafd801d4af343f02e4ffbfe/docs/metrics/README.md?plain=1#L89

**Review by Kielek — APPROVED** [review]
Q: Do you know if it should trigger any changes on instrumentation (packages) side?
Contrib repository/natively instrumented packages?

File: test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs [hunk 1]
```diff
@@ -1613,6 +1613,34 @@ public void UnsupportedMetricInstrument()
...
+        Assert.Single(exportedItems);
+        var metric = exportedItems[0];
+        Assert.Equal("Background Noise Level", metric.Description);
+        List<MetricPoint> metricPoints = new List<MetricPoint>();
+        foreach (ref readonly var mp in metric.GetMetricPoints())
+        {
+            metricPoints.Add(mp);
+        }
+
+        var histogramPoint = metricPoints[0];
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs:1639]:
```suggestion
        var gaugePoint = metricPoints[0];
```

File: test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs [hunk 2]
```diff
@@ -1613,6 +1613,34 @@ public void UnsupportedMetricInstrument()
         Assert.Empty(exportedItems);
     }
 
+    [Fact]
+    public void GaugeIsExportedCorrectly()
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs:1617]: Please expand the test to cover both temporalities. Though Gauge has no temporality notion like other instruments, but still temporality affects if previous reported metricpoints are remembered/forgot after collect.

In particular, we need a test to validate that
For sync gauge, if cumulative temporality is set, metrics are exported every collect, irrespective of whether new measurements were reported or not.

**Review by cijothomas — CHANGES_REQUESTED** [review]
Change looks good.
Requesting changes to ensure the test covers a corner case with Gauge and Temporality, see https://github.com/open-telemetry/opentelemetry-dotnet/pull/5867/files#r1778773598

File: test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs (hunk 2 above)
Comment (cijothomas) [reply test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs:1617]: Relevant spec wording : https://github.com/open-telemetry/opentelemetry-specification/pull/3540/files#diff-32d5f289beed96900fa1e6e69a3ee9cc0b6503ce8883672011c44bc772c0fa2aR1099

Comment (rajkumar-rangaraj) [reply test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs:1617]: Fixed, please check.

**Comment by rajkumar-rangaraj** [comment]
> Q: Do you know if it should trigger any changes on instrumentation (packages) side? Contrib repository/natively instrumented packages?

Not yet, need to investigate it.

File: test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs [hunk 3]
```diff
@@ -1613,6 +1613,98 @@ public void UnsupportedMetricInstrument()
...
+                secondMetricPoints.Add(metricPoint);
+            }
+
+            Assert.Single(secondMetricPoints);
+            var secondMetricPoint = secondMetricPoints[0];
+            Assert.Equal(10, secondMetricPoint.GetGaugeLastValueLong());
+        }
+        else if (temporalityPreference == MetricReaderTemporalityPreference.Delta)
+        {
+            // For delta temporality, no new metric should be collected
```
Comment (cijothomas) [inline test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs:1697]: can we can check exportedMetrics's length, instead of relying on Exception?

**Review by cijothomas — APPROVED** [review]

**Comment by J4M35K** [comment]
Can we please get this merged @rajkumar-rangaraj ?
